test/e2e/instrumentation/monitoring/custom_metrics_stackdriver.go

```diff
@@ -48,41 +49,46 @@ var _ = instrumentation.SIGDescribe("Stackdriver Monitoring", func() {
 	})
 
 	f := framework.NewDefaultFramework("stackdriver-monitoring")
-	var kubeClient clientset.Interface
-	var customMetricsClient customclient.CustomMetricsClient
-	var discoveryClient *discovery.DiscoveryClient
 
-	It("should run Custom Metrics - Stackdriver Adapter [Feature:StackdriverCustomMetrics]", func() {
-		kubeClient = f.ClientSet
+	It("should run Custom Metrics - Stackdriver Adapter for old resource model [Feature:StackdriverCustomMetrics]", func() {
+		kubeClient := f.ClientSet
 		config, err := framework.LoadConfig()
 		if err != nil {
 			framework.Failf("Failed to load config: %s", err)
 		}
-		customMetricsClient = customclient.NewForConfigOrDie(config)
-		discoveryClient = discovery.NewDiscoveryClientForConfigOrDie(config)
-		testAdapter(f, kubeClient, customMetricsClient, discoveryClient)
+		customMetricsClient := customclient.NewForConfigOrDie(config)
+		discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(config)
+		testCustomMetricsOldResourceModel(f, kubeClient, customMetricsClient, discoveryClient)
+	})
+
+	It("should run Custom Metrics - Stackdriver Adapter for new resource model [Feature:StackdriverCustomMetrics]", func() {
+		kubeClient := f.ClientSet
+		config, err := framework.LoadConfig()
+		if err != nil {
+			framework.Failf("Failed to load config: %s", err)
+		}
+		customMetricsClient := customclient.NewForConfigOrDie(config)
+		discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(config)
+		testCustomMetricsNewResourceModel(f, kubeClient, customMetricsClient, discoveryClient)
+	})
+
+	It("should run Custom Metrics - Stackdriver Adapter for external metrics [Feature:StackdriverExternalMetrics]", func() {
+		kubeClient := f.ClientSet
+		config, err := framework.LoadConfig()
+		if err != nil {
+			framework.Failf("Failed to load config: %s", err)
+		}
+		externalMetricsClient := externalclient.NewForConfigOrDie(config)
+		testExternalMetrics(f, kubeClient, externalMetricsClient)
 	})
 })
 
-func testAdapter(f *framework.Framework, kubeClient clientset.Interface, customMetricsClient customclient.CustomMetricsClient, discoveryClient *discovery.DiscoveryClient) {
+func testCustomMetricsNewResourceModel(f *framework.Framework, kubeClient clientset.Interface, customMetricsClient customclient.CustomMetricsClient, discoveryClient *discovery.DiscoveryClient) {
```

Comment:
Is there any difference between `testCustomMetricsNewResourceModel` and `testCustomMetricsOldResourceModel` other than which adapter they use? It feels like we're duplicating a lot of code that we don't need to duplicate.

Reply:
This is the only difference, I've removed duplicated code.